src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -142,6 +140,7 @@ internal sealed class AggregatorStore
         }
         else
         {
+            this.tagsToMetricPointIndexDictionary = new(concurrencyLevel: Environment.ProcessorCount, capacity: lookupCapacity);
```

Comment:
is this related to the changes in Tags in anyway? If not, I'd suggest to keep it in a separate PR, so we can see what is the effect of that change in perf independently.

Reply:
It won't show on a benchmark, I don't think. I'd have to construct something whacky/custom to prove it eliminates the growth pause. Is that what are looking to see? My thinking is... we know there is growth penalty when [Concurrent]Dictionary has to expand. This change is to remove that jitter. I had a conversation with @reyang which was about people not understanding how to properly measure performance and might misinterpret that jitter as unstable SDK. Something like that, paraphrasing from memory. We do reserve the capacity when using delta reclaim today. This just does that for everything.

What I'm saying is... doesn't seem like a good use of time to split it out and try to prove in a benchmark it is useful. Ipso facto argument more or less. Happy to do it though if you _really_ want it.

Reply:
I didn't mean to say - we need to benchmark this particular use case. I agree with the direction of pre-allocating enough memory ahead of time. We do that wherever feasible today, and this is just filling one more gap. So I agree with this change, _but_ I always prefer to keep its own separate PR, so its clear why are we doing it, and also to show that - the gains shows in this PR is coming due to changes in Tags (hash/equals), not with pre-allocating dictionary's internal array aptly!